pipeline_dp/accumulator.py

```diff
@@ -257,9 +264,34 @@ def compute_metrics(self):
         return dp_computations.add_noise_vector(self._vec_sum, self._params)
 
 
-@dataclass
 class SumParams:
-    noise: dp_computations.MeanVarParams
+    """Parameters for a sum accumulator.
+
+    Wraps epsilon and delta from the budget which are lazily loaded.
+    AggregateParams are copied into a MeanVarParams instance.
+    """
+
+    def __init__(self, budget: pipeline_dp.budget_accounting.MechanismSpec,
+                 aggregate_params: aggregate_params.AggregateParams):
+        self._budget = budget
+        self._aggregate_params = aggregate_params
```

Comment:
Thanks, I like this approach more that with inheritance!

Reply:
Awesome, it seems to work well. Thanks!